meta-mbl-distro/scripts/create-update-payload

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+"""
+This script will create a bootloader update payload tar archive.
+
+You can specify a compression type for the tar archive. The default is gz and
+the available options are gz, bz2 and xz.
+
+To use the script you can either pass in a space separated list of file names
+or a list of bootloader slot numbers. We then map the file names of the update
+components to the corresponding bitbake variables, which are named according
+to the bootloader slot the file will end up in.
+
+We rename the files MBL_WKS_BOOTLOADER[n] so the update scripts on the target
+only have to know which bootloader slot they're updating, and are generic for
+any MACHINE target.
+
+The script assumes the files you want to archive are in DEPLOY_DIR_IMAGE in
+your MBL workspace.
+
+The script supports multi-component or single component payloads.
+
+Example invocation:
+
+create-update-payload --bootloader-slots 1 2 --output-tar-path tmp/payload.tar
+
+This will create the update payload containing the bootloader 1 and 2
+components at tmp/payload.tar
+"""
+
+import pathlib
+import re
+import shutil
+import sys
+import tarfile
+import tempfile
+
+from itertools import product
+
+# Set the PYTHONPATH so we can import openembedded scripts/lib modules.
+# This script can only be executed from the
+# meta-mbl-distro/scripts folder in the MBL workspace.
```

Comment:
I think the note about from where the script can be executed should be in the file's docstring along with the sample invocation.

Reply:
...But hang on, isn't `__file__` (used below) the path to the current Python file? Then can't we just work out where things are relvative to this file rather than saying we must execute the script from a particular directory?

Reply:
The comment is wrong, we can execute the script from anywhere, but we can't move the script itself from the folder it's currently in as that will screw up the import paths. The problem is the script needs to import stuff from openembedded core to use the bitbake and oe libs. I'll move the comment to the docstring and correct it.